ui/apps/vue-mri-ui-lib/src/query-filter/components/GroupCriteriaMenu.vue

```diff
@@ -0,0 +1,322 @@
+    <template #default="{ hide }">
+      <div class="popover-content">
+        <div class="group-criteria-menu">
+          <div class="body">
+            <div class="group-criteria-menu__container">
+              <!-- Button Layout Container -->
+              <div class="button-layout-container">
+                <!-- Segmented Button Group -->
+                <div class="segmented-button-group">
+                  <button
+                    class="segment-button segment-button--all"
+                    :class="{ 'segment-button--selected': isActiveGroupCriteriaType('ALL') }"
+                    @click="updateActiveGroupCriteriaType('ALL')"
+                  >
+                    All
+                  </button>
+                  <button
+                    class="segment-button segment-button--any"
+                    :class="{ 'segment-button--selected': isActiveGroupCriteriaType('ANY') }"
+                    @click="updateActiveGroupCriteriaType('ANY')"
+                  >
+                    Any
+                  </button>
+                  <button
+                    class="segment-button segment-button--at-least"
+                    :class="{ 'segment-button--selected': isActiveGroupCriteriaType('AT_LEAST') }"
+                    @click="updateActiveGroupCriteriaType('AT_LEAST')"
+                  >
+                    At least
+                  </button>
+                  <button
+                    class="segment-button segment-button--at-most segment-button--last"
+                    :class="{ 'segment-button--selected': isActiveGroupCriteriaType('AT_MOST') }"
+                    @click="updateActiveGroupCriteriaType('AT_MOST')"
+                  >
+                    At most
+                  </button>
+                </div>
+                
+                <!-- Count Dropdown - Always reserve space -->
+                <div class="count-dropdown-container">
+                  <div 
+                    v-show="showCountDropdown" 
+                    class="count-dropdown" 
+                    ref="countDropdownRef"
+                  >
+                    {{ currentCount }}
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+          
+          <div class="footer">
+            <ButtonMaterial
+              @button-click="
+                () => {
+                  updateGroupCriteriaField()
+                  hide()
+                }
+              "
+            >
+              OK
+            </ButtonMaterial>
+          </div>
+        </div>
+      </div>
+    </template>
+  </Popper>
+
+  <!-- Count dropdown menu -->
+  <DropdownMenu
+    v-if="countDropdownRef && showCountDropdown"
+    :options="occurrenceCountOptions"
+    @select="(value: string) => updateCountState(value)"
+    :target="countDropdownRef"
+  />
+</template>
+
+<style lang="scss">
```

Comment:
if possible, use 
```suggestion
<style scoped lang="scss">
```
if global scope is needed in the case of overwriting some styles from some libraries, we can have both
```
<style scoped lang="scss">

and 

<style lang="scss">
```
coexist in the file

Reply:
Got it. Thanks